src/report/report.service.ts

```diff
@@ -30,27 +33,33 @@ export class ReportService {
     return ReportService.docToPublicReport(report);
   }
 
-  async getReportStatistics(): Promise<ReportStatisticsDto> {
-    const stats = await this.reportRepository.getVisibleStatusCounts();
+  async getReportStatistics(
+    category?: ReportCategory,
+  ): Promise<ReportStatisticsDto> {
+    const stats = await this.reportRepository.getVisibleStatusCounts(category);
     return ReportService.docToReportStatistics(stats);
   }
 
   private static docToReportStatistics(e: any): ReportStatisticsDto {
     return new ReportStatisticsDto(
-      e.filter((stat: { _id: string }) => stat._id == 'gautas')[0].count ?? 0,
-      e.filter((stat: { _id: string }) => stat._id == 'tiriamas')[0].count ?? 0,
-      e.filter((stat: { _id: string }) => stat._id == 'ištirtas')[0].count ?? 0,
-      e.filter((stat: { _id: string }) => stat._id == 'sutvarkyta')[0].count ??
-        0,
-      e.filter((stat: { _id: string }) => stat._id == 'nepasitvirtino')[0]
-        .count ?? 0,
+      ReportService.filterStatistics(e, 'gautas'),
+      ReportService.filterStatistics(e, 'tiriamas'),
+      ReportService.filterStatistics(e, 'ištirtas'),
+      ReportService.filterStatistics(e, 'sutvarkyta'),
+      ReportService.filterStatistics(e, 'nepasitvirtino'),
     );
   }
 
+  private static filterStatistics(e: any, status: string): number {
+    return e.filter((stat: { _id: string }) => stat._id == status).length > 0
+      ? e.filter((stat: { _id: string }) => stat._id == status)[0].count ?? 0
+      : 0;
```

Comment:
Let's keep it DRY. Also, consider utilizing Array.at (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at), which gracefully handles undefined values for non-existent array indices